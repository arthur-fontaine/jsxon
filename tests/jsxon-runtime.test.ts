import { test, describe, expect } from "vitest";
import createJiti from "jiti";
import * as babel from "@babel/core";
import transformPlugin from "@babel/plugin-transform-react-jsx";

const jiti = createJiti(import.meta.url);

describe("jsxon runtime", () => {
	test("should return a json object describing the element", () => {
		const code = /* tsx */ `
      import JSXon, { Text } from '../src/index.ts'
      export const element = <Text text="Hello, JSXon!" />
    `;
		const result = babel.transformSync(code, {
			plugins: [
				[
					transformPlugin,
					{
						pragma: "JSXon.createElement",
						pragmaFrag: "JSXon.Fragment",
					},
				],
			],
		});

		const transformedCode = result?.code;
		const executedCode =
			transformedCode &&
			jiti.evalModule(transformedCode, {
				ext: ".ts",
				id: import.meta.url,
				filename: import.meta.url,
			});

		expect(executedCode).toHaveProperty("element");
		expect((executedCode as any).element).toMatchSnapshot();
	});
});
