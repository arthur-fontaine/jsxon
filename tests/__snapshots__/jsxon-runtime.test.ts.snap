// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`jsxon runtime > should return a json object describing the element 1`] = `
{
  "props": {
    "text": "Hello, JSXon!",
  },
  "type": "jsxon:Text",
}
`;
